src/js_command_match/contract/call.rs

```diff
@@ -0,0 +1,179 @@
+use near_gas::NearGas;
+
+use crate::js_command_match::constants::{
+    DEFAULT_SEED_PHRASE_PATH, LEDGER_PATH_ALIASES, NETWORK_ID_ALIASES, SIGN_WITH_LEDGER_ALIASES,
+    USE_ACCOUNT_ALIASES,
+};
+
+#[derive(Debug, Clone, clap::Parser)]
+pub struct CallArgs {
+    contract_name: String,
+    method_name: String,
+    #[clap(default_value = "")]
+    args: String,
+    #[clap(long, aliases = USE_ACCOUNT_ALIASES)]
+    use_account: String,
+    #[clap(long, aliases = SIGN_WITH_LEDGER_ALIASES, default_value_t = false)]
+    sign_with_ledger: bool,
+    #[clap(long, aliases = LEDGER_PATH_ALIASES, default_value = Some(DEFAULT_SEED_PHRASE_PATH))]
+    ledger_path: Option<String>,
+    #[clap(long, aliases = NETWORK_ID_ALIASES, default_value=None)]
+    network_id: Option<String>,
+    #[clap(long, default_value_t = 30_000_000_000_000)]
+    gas: u64,
+    #[clap(long, default_value = "0")]
+    deposit: String,
+    #[clap(long, default_value = "0", conflicts_with = "deposit", aliases = ["depositYocto"])]
+    deposit_yocto: String,
+    #[clap(long, default_value_t = false)]
+    base64: bool,
+    #[clap(long, aliases = ["privateKey"], default_value=None)]
+    private_key: Option<String>,
+}
+
+impl CallArgs {
+    pub fn to_cli_args(&self, network_config: String) -> Vec<String> {
+        let network_id = self.network_id.clone().unwrap_or(network_config);
+
+        let mut command = vec![
+            "contract".to_string(),
+            "call-function".to_string(),
+            "as-transaction".to_string(),
+            self.contract_name.to_owned(),
+            self.method_name.to_owned(),
+        ];
+
+        if self.base64 {
+            command.push("base64-args".to_string());
+        } else {
+            command.push("json-args".to_string());
+        };
+
+        command.push(self.args.to_owned());
+        command.push("prepaid-gas".to_string());
+        command.push(format!("{} Tgas", NearGas::from_gas(self.gas).as_tgas()));
+        command.push("attached-deposit".to_string());
+
+        if self.deposit_yocto != "0" {
+            command.push(format!("{} yoctonear", self.deposit_yocto));
+        } else {
+            command.push(format!("{} NEAR", self.deposit));
+        }
+
+        command.push("sign-as".to_string());
+        command.push(self.use_account.to_owned());
+        command.push("network-config".to_string());
+        command.push(network_id);
+
+        if self.sign_with_ledger {
+            command.push("sign-with-ledger".to_string());
+            command.push("--seed-phrase-hd-path".to_string());
+            command.push(self.ledger_path.to_owned().unwrap_or_default());
+        }
+
+        if self.private_key.is_some() {
+            command.push("sign-with-plaintext-private-key".to_string());
+            command.push(self.private_key.clone().unwrap());
+        }
+
+        if !self.sign_with_ledger && self.private_key.is_none() {
+            command.push("sign-with-keychain".to_string());
+        }
```

Comment:
nit: Consider using `else` and `if let` instead of `if` + `.unwrap()`:

```suggestion
        if self.sign_with_ledger {
            command.push("sign-with-ledger".to_string());
            command.push("--seed-phrase-hd-path".to_string());
            command.push(self.ledger_path.to_owned().unwrap_or_default());
        } else if let Some(private_key) = self.private_key {
            command.push("sign-with-plaintext-private-key".to_string());
            command.push(private_key.clone());
        } else {
            command.push("sign-with-keychain".to_string());
        }
```